### PR TITLE
Remove parsing catalyst device attribute dictionary for shots

### PR DESCRIPTION
**Pull request by paul0403**
**Context:**
As part of the work to support dynamic measurement shapes, https://github.com/PennyLaneAI/catalyst/pull/1310 in Catalyst is allowing a quantum device to take in a dynamic number of shots at the mlir level. One of the changes involved is that the `DeviceInitOp` now takes in a proper SSA argument for shots, instead of having shots as just another entry in the `DeviceInitOp`'s attribute dictionary.

Correspondingly, the backend devices' catalyst interfaces need to stop parsing the `DeviceInitOp`'s attribute dictionary for shots.

**Description of the Change:**
The backend devices' catalyst interfaces stop parsing the `DeviceInitOp`'s attribute dictionary for shots.

**Benefits:**
Agreement with Catalyst on how device shots, potentially dynamic, is handled.

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Comment by paul0403** [comment]
do not merge until catalyst merges https://github.com/PennyLaneAI/catalyst/pull/1310

**Review by maliasadi — COMMENTED** [review]
LGTM! @paul0403  there are still a few steps to be done:
- remove the C++ tests for device_shots
- update the code formatting
- update the changelog

**Comment by paul0403** [comment]
The parsing has the following purpose: if the shots attribute exists in the dictionary, set device shots to the value seen; if shots attribute is not in the dictionary then set shots to zero.

By defaulting the `device_shots` to zero, we achieve the same effect.

The setting of `device_shots` now occurs in [`__catalyst__rt__device_init`](https://github.com/PennyLaneAI/catalyst/blob/main/runtime/lib/capi/RuntimeCAPI.cpp#L256). Since the cpp tests in lightning repo do not go through catalyst capi, giving the `device_shots` a zero value default should be enough to pass all tests. The tests originally get zero shots because their devices are initialized with a default empty attribute dictionary, which gets parsed to zero shots. @maliasadi

File: pennylane_lightning/core/src/simulators/lightning_qubit/catalyst/tests/Test_LightningDriver.cpp
Comment (paul0403) [reply pennylane_lightning/core/src/simulators/lightning_qubit/catalyst/tests/Test_LightningDriver.cpp:1]: The two links I had above had seemingly different classes, one called `LQSimulator` and the other called `LightningSimulator`, so I was just pointing out that these are really the same class

File: pennylane_lightning/core/_version.py
Comment (paul0403) [inline pennylane_lightning/core/_version.py:19]: changelog

**Review by erick-xanadu — APPROVED** [review]
Looks good!

File: pennylane_lightning/core/src/simulators/lightning_qubit/catalyst/tests/Test_LightningDriver.cpp
Comment (maliasadi) [reply pennylane_lightning/core/src/simulators/lightning_qubit/catalyst/tests/Test_LightningDriver.cpp:1]: I don't know why we have this alias for Catalyst/LQ tests either.. @rauletorresc kept the Catalyst style for multiple lightning devices when downstreaming LQ. There's no problem with that. It's fewer letters to type :smile:

**Review by maliasadi — APPROVED** [review]
LGTM! Thanks @paul0403

**Pull request merged**
